effect/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -701,10 +701,10 @@ private object RTS {
 
           if (finalizer == null) {
             if (stack.isEmpty()) done(value.asInstanceOf[Try[A]])
-            else continueWithValue(value, \/-(value))
+            else continueWithValue(value, value)
```

Comment:
Here I'm less sure if we should continue with `value` (which is already a `Try`), or rather with a `\/-(t)` (as we now value is a failure).
However, the former makes the test pass, and the latter results in a ClassCastException, so I think this is also correct.